From a90ccc240501bf3362b23f67771f65b7dec2ccf9 Mon Sep 17 00:00:00 2001
From: Jamie Madill <jmadill@chromium.org>
Date: Fri, 29 Jul 2022 14:26:37 -0400
Subject: [PATCH] Remove default copy constructor in header. (#4879)

A recent libc++ roll in Chrome warned of a deprecated copy. We're
still looking if this is a bug in libc++ or a valid warning, but
removing the redundant line is a safe workaround or fix in either
case.

See discussion in https://crrev.com/c/3791771

Upstream-Status: Backport [https://github.com/KhronosGroup/SPIRV-Tools/pull/4879]
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 source/opt/merge_return_pass.h | 2 --
 1 file changed, 2 deletions(-)

diff --git a/source/opt/merge_return_pass.h b/source/opt/merge_return_pass.h
index a35cf269..d15db2f6 100644
--- a/source/opt/merge_return_pass.h
+++ b/source/opt/merge_return_pass.h
@@ -118,8 +118,6 @@ class MergeReturnPass : public MemPass {
     StructuredControlState(Instruction* break_merge, Instruction* merge)
         : break_merge_(break_merge), current_merge_(merge) {}
 
-    StructuredControlState(const StructuredControlState&) = default;
-
     bool InBreakable() const { return break_merge_; }
     bool InStructuredFlow() const { return CurrentMergeId() != 0; }
 
-- 
2.37.2

